peachjam/views/generic_views.py

```diff
@@ -43,12 +45,25 @@ class ClampedPaginator(Paginator):
     * https://github.com/photocrowd/django-cursor-pagination
     """
 
+    def __init__(self, *args, **kwargs):
+        self.page_title = kwargs.pop("page_title")
+        super().__init__(*args, **kwargs)
+
     max_num_pages = 10
 
     @cached_property
     def num_pages(self):
         return min(super().num_pages, self.max_num_pages)
 
+    @cached_property
+    def count(self):
+        cache_key = f"paginator_count_{slugify(self.page_title)}"
```

Comment:
surely the request.path (or including the query string params) is a more consistent cache key?